### PR TITLE
TFJob application CR

### PR DESCRIPTION
**Pull request by richardsliu**
Fixes https://github.com/kubeflow/manifests/issues/119

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/142)

### CODE REVIEW

**Comment by hougangliu** [comment]
@richardsliu please hold it util we make it clear by discuss in https://github.com/kubeflow/kubeflow/pull/3429 about how to enable applications for a package
Thanks

File: tf-training/tf-job-operator/base/application.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  type: "tf-job-operator"
+  componentKinds:
+    - group: core
+      kind: Service
+    - group: apps
+      kind: Deployment
+    - group: core
+      kind: ConfigMap
+    - group: kubeflow.org
+      kind: TFJob
```
Comment (hougangliu) [inline tf-training/tf-job-operator/base/application.yaml]: should we also include other resource, such as clusterrole, clusterrolebinding, crd and so on?

Comment (richardsliu) [reply tf-training/tf-job-operator/base/application.yaml]: Good question, I was wondering this as well. I did not see clusterRole/bindings/serviceAccounts added in https://github.com/kubeflow/manifests/pull/105.

Comment (hougangliu) [reply tf-training/tf-job-operator/base/application.yaml]: I think for crd, maybe we will need another application.yaml based on current orginazation

File: tf-training/tf-job-operator/base/application.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+apiVersion: app.k8s.io/v1beta1
```
Comment (kkasravi) [inline tf-training/tf-job-operator/base/application.yaml]: Hi Richard

Could you create tf-training/tf-job-operator/overlays/application/{kustomization.yaml, application.yaml}. You can look at jupyter/notebook-controller/overlays/application

**Comment by richardsliu** [comment]
@kkasravi Made the changes, could you take a look?

**Pull request merged**